src/MassiveAction.php

```diff
@@ -1124,52 +1124,47 @@ public static function showMassiveActionsSubForm(MassiveAction $ma)
                     $search_options = [];
                 }
 
-                $items         = [];
+                // TODO: ensure that all items are equivalent ...
+                $item   = null;
+                $search = null;
                 foreach ($search_options as $search_option) {
                     $search_option = explode(':', $search_option);
-                    $itemtype      = $search_option[0];
-                    $index         = $search_option[1];
+                    $so_itemtype   = $search_option[0];
+                    $so_index      = $search_option[1];
 
-                    if (!$item = getItemForItemtype($itemtype)) {
+                    if (!$so_item = getItemForItemtype($so_itemtype)) {
                         continue;
                     }
 
-                    if (Infocom::canApplyOn($itemtype)) {
-                        Session::checkSeveralRightsOr([$itemtype  => UPDATE,
+                    if (Infocom::canApplyOn($so_itemtype)) {
+                        Session::checkSeveralRightsOr([$so_itemtype  => UPDATE,
                             "infocom"  => UPDATE
                         ]);
                     } else {
-                        $item->checkGlobal(UPDATE);
+                        $so_item->checkGlobal(UPDATE);
                     }
 
-                    $search = Search::getOptions($itemtype);
-                    if (!isset($search[$index])) {
-                         exit();
+                    $itemtype_search_options = Search::getOptions($so_itemtype);
+                    if (!isset($itemtype_search_options[$so_index])) {
+                       exit();
                     }
-                    $item->search = $search[$index];
-
-                    $items[] = $item;
-                }
 
-                if (count($items) == 0) {
-                    exit();
+                    $item   = $so_item;
+                    $search = $itemtype_search_options[$so_index];
+                    break; // No need to process all items a corresponding item/searchoption has been found
                 }
 
-                // TODO: ensure that all items are equivalent ...
-                $item   = $items[0];
```

Comment:
Unless i miss something, there is an additional usage of `$item->` below (old line: 1197)

```
echo $item->getValueToSelect($search, $fieldname, $values, $options);
```

Reply:
`$item` is defined in line 1152.

Previous code was creating an array of `$items` to get only the first one. New logic just store the first `$item` in the var directly, and the stop the loop.